docs/book/beyond_basics/what_is_a_webhook.md

```diff
@@ -0,0 +1,81 @@
+# Webhook
+
+Webhooks are HTTP callbacks, providing a way for notifications to be delivered to an external web server.
+A web application implementing webhooks will send an HTTP request (typically POST) to other application when certain event happens. 
+In kubernetes world, there are 3 kinds of webhooks: admission webhook, authorization webhook and CRD conversion webhook.
+
+## Admission Webhook
+
+Admission webhooks are HTTP callbacks that receive admission requests, process them and return admission responses.
+There are two types of admission webhooks: mutating admission webhook and validating admission webhook.
+With mutating admission webhooks, you may change the requests to enforce custom policy, e.g. injecting a sidecar container
+With validating admission webhooks, you may reject the requests to enforce custom admission policies, e.g. image whitelisting.
+
+#### Why Admission Webhook is Important
+
+Admission webhook is the mechanism to enable kubernetes extensibility through CRD.
+- Mutating admission webhook is the only way to do defaulting for CRDs.
+- Validating admission webhook is the only to do CRD validation that is not covered by per field basis OpenAPI validation.
+e.g. cross-field validation in one object or cross-object validation.
+
+It can also be used to add custom logic in the core kubernetes API.
+
+#### Mutating Admission Webhook
+
+A mutating admission webhook receives an admission request which contains an object.
+The webhook can either decline the request directly or returning JSON patches for modifying the original object.
+- If admitting the request, the webhook is responsible for generating JSON patches and send them back in the 
+admission response.
+- If declining the request, a reason message should be returned in the admission response.
+
+#### Validating Admission Webhook
+
+A validating admission webhook receives an admission request which contains an object.
+The webhook can either admit or decline the request.
+A reason message should be returned in the admission response if declining the request.
+
+#### Authentication
+
+The apiserver by default doesn't authenticate itself to the webhooks.
```

Comment:
Is this the auth for the requesting user, or the apiserver itself?

Reply:
The apiserver itself